PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['account.fiscal.position']
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']

Total Changes: 8

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/base_vat/models/account_fiscal_position.py
+++ b/addons/base_vat/models/account_fiscal_position.py
@@ -5,7 +5,7 @@ from odoo.exceptions import ValidationError
 
 
 class AccountFiscalPosition(models.Model):
-    _inherit = ['account.fiscal.position']
+    _inherit = 'account.fiscal.position'
 
     @api.model_create_multi
     def create(self, vals_list):

--- a/addons/base_vat/models/res_company.py
+++ b/addons/base_vat/models/res_company.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     vat_check_vies = fields.Boolean(string='Verify VAT Numbers')

--- a/addons/base_vat/models/res_config_settings.py
+++ b/addons/base_vat/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     vat_check_vies = fields.Boolean(related='company_id.vat_check_vies', readonly=False,
         string='Verify VAT Numbers')

--- a/addons/base_vat/models/res_partner.py
+++ b/addons/base_vat/models/res_partner.py
@@ -87,7 +87,7 @@ _region_specific_vat_codes = {
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     vies_valid = fields.Boolean(
         string="Intra-Community Valid",
